apps/keyboard/js/settings/layout_item_view.js

```diff
@@ -53,34 +53,34 @@ LayoutItemView.prototype.start = function() {
 };
 
 LayoutItemView.prototype._updateProgress = function() {
-  var loadedSizeData = this._getHumanFileSize(this._model.downloadLoadedSize);
-  var totalSizeData = this._getHumanFileSize(this._model.downloadTotalSize);
   this._progressEl.value = this._model.downloadLoadedSize;
   this._progressEl.max = this._model.downloadTotalSize;
-  this._statusEl.dataset.l10nArgs =
-    JSON.stringify({
-      loadedSize: loadedSizeData.size,
-      loadedSizeUnit: loadedSizeData.localizedUnit,
-      totalSize: totalSizeData.size,
-      totalSizeUnit: totalSizeData.localizedUnit
-    });
+
+  Promise.all([
+    this._getHumanFileSize(this._model.downloadLoadedSize),
+    this._getHumanFileSize(this._model.downloadTotalSize)
+  ]).then(([loadedSize, totalSize]) => {
```

Comment:
es6 will unfortunately break Keyboard demo running on the Web, which runs keyboard with code in the source tree. Could you use `values[0]` and `values[1]` here?